src/entities/Resource.js

```diff
@@ -113,4 +123,46 @@ export default class Resource {
   getTags() {
     return this.tags;
   }
+
+  /**
+   * Highlight all appearances of keyword in item.
+   * E.x: the keyword 'sa' will be highlighted 2 times in string 'saritasa/api'
+   *
+   * @param {string} keyword - Search keyword
+   */
+  addHighlight(keyword) {
+    if (!keyword) {
+      return this;
+    }
+
+    const highlightedRes = _.clone(this);
+    const sentences = [ this.getName(), this.getDesc(), this.getExplanation() ];
+
+    _.forEach(sentences, (sen, idx) => {
+      if (sen === undefined) {
+        return;
+      }
+
+      let text = sen;
+      let offset = 0;
+      let index = text.toLowerCase().indexOf(keyword.toLowerCase(), offset);
+
+      while (index >= 0) {
+        sentences[idx] = text.substring(0, index);
+        sentences[idx] += '<span class="highlighted-word">';
```

Comment:
Like I'm said before I think it should be in different place.

Reply:
Agree, entity (model - M in **M**VC) should not contain any markup (presentation - how data is presented to user). Template (View in M**V**C) must be responsible for that.

I think, that easy solution - is to put words, being searched to some global context (application state, storage).
Then in template markup markup you may place something like that:
```
<span>{{ description | highlightSearchTerms }}<span>
```